account_move_stripe_import/models/account_move.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2020 Akretion (https://www.akretion.com).
+# @author Pierrick Brun <pierrick.brun@akretion.com>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import _, fields, models
+from odoo.addons.account_move_base_import.models.account_move import (
+    ErrorTooManyPartner,
+)
+
+
+class AccountMoveCompletionRule(models.Model):
+    """Add a rule based on transaction ID"""
+
+    _inherit = "account.move.completion.rule"
+
+    function_to_call = fields.Selection(
+        selection_add=[
+            (
+                "get_from_transaction_ref_and_so",
+                "Match Sales Order using transaction Reference",
+            ),
+        ]
+    )
+
+    def get_from_transaction_ref_and_so(self, line):
+        """
+        Same as get_from_transaction_id_and_so but uses the reference field
+        of the transaction instead of its ID.
+        """
+        res = {}
+        so_obj = self.env["sale.order"]
+        reference = line.name
+        if line.name.startswith("REFUND"):
+            # Substring between
+            reference = line.name[len("REFUND FOR CHARGE(")+1:-len(")")]
+        sales = so_obj.search([("transaction_ids.reference", "=", reference)])
```

Comment:
With Odoo10 Website E-commerce, when the customer choose a payment acquirer (for example, wire transfer) and then change his mind and choose another payment acquirer like stripe, then we end up with two transactions in the database. The second one's name is the sale order + "x1".

![image](https://user-images.githubusercontent.com/61087258/117415572-c1c63380-af18-11eb-9602-3b2ab0c3dee0.png)

Also, we sometimes have the stripe transaction id in line.name (it is the case in belgium with stripe bancontact payments).

This would solve theses two cases : 
sales = so_obj.search(['|',
                              ("payment_tx_id.reference", "=like", reference+'%'),
                              ("payment_tx_id.acquirer_reference", "=", reference)])